swingset/src/main/java/com/nqadmin/swingset/SSComboBox.java

```diff
@@ -831,33 +870,38 @@ public void setSelectedValue(final Integer _value) {
 	 */
 	@Override
 	public void updateSSComponent() {
-		
-		// TODO Modify this class similar to updateSSComponent() in SSFormattedTextField and only allow JDBC types that convert to Long or Integer
-		
+		// TODO Modify this class similar to updateSSComponent() in SSFormattedTextField and only limit JDBC types accepted
 		try {
+			// If initialization is taking place then there won't be any mappings so don't try to update anything yet.
 			if (!hasItems()) {
 				return;
 			}
 
-			// Expecting an integer so trim in case the database column is a String AND has padding
-			// TODO: #17 string casting
-			final String rawText = getBoundColumnText();
-			logger.debug("{}: getBoundColumnText() returns {}.", () -> getColumnForLog(), () -> rawText);
-			final String trimmedText = rawText != null ? rawText.trim() : null;
-
-			// GET THE INTEGER EQUIVALENT OF THE TEXT IN THE TEXT FIELD
-			if ((trimmedText != null) && !(trimmedText.isEmpty())) {
-				final int comboCode = Integer.parseInt(trimmedText);
+			// SSDBComboBox will generally work with primary key column data queried from the database, which will generally be of data type long.
+			// SSComboBox is generally used with 2 or 4 byte integer columns.
+			final String boundColumnText = getBoundColumnText();
 
-				setSelectedMapping(comboCode);
+			// LOGGING
+			logger.debug("{}: getBoundColumnText() - " + boundColumnText, () -> getColumnForLog());
 
-			} else {
-				//setSelectedIndex(-1);
+			// GET THE BOUND VALUE STORED IN THE ROWSET - may throw a NumberFormatException
+			Integer targetValue = null;
+			if ((boundColumnText != null) && !boundColumnText.isEmpty()) {
+				targetValue = Integer.parseInt(boundColumnText);
+			}
+			
+			// LOGGING
+			logger.debug("{}: targetValue - " + targetValue, () -> getColumnForLog());
+			
+			// UPDATE COMPONENT
+			if (targetValue==null) {
 				setSelectedItem(nullItem);
+			} else {
+				setSelectedMapping(targetValue);
 			}
```

Comment:
Why not just `setSelectedMapping(targetValue)`?

Reply:
Maybe that would be OK. If getAllowNull()==false, does setSelectedMapping(null) work OK?

Reply:
> does setSelectedMapping(null) work OK?

It seems like it after a quick look. But it's hardly an important issue.

Reply:
Changed it and it seems to work fined. Committed/pushed updates to SSComboBox and SSDBComboBox.